src/main/java/newbank/server/commands/DepositCommand.java

```diff
@@ -25,36 +29,32 @@ protected String getSyntax() {
 
   @Override
   public String execute() {
-    if ((tokens.length >= 2 && tokens[1].equalsIgnoreCase("HELP"))) {
-      return String.format("SUCCESS: Usage: %s", getSyntax());
-    }
-
-    try {
-      checkLoggedIn(customer);
-    } catch (RequestNotAllowedException ex) {
-      return String.format("FAIL: %s", ex.getMessage());
-    }
-
-    if (tokens.length != 3) {
-      return String.format("FAIL: Usage: %s", getSyntax());
-    }
+    Optional<String> message = applyResponsibilityChain();
+    if (message.isPresent()) return message.get();
 
     final String accountName = tokens[1];
 
     try {
-      final Double amount = Double.parseDouble(tokens[2]);
-
-      if (!(amount > 0)) {
-        return String.format("FAIL: Deposit amount [%s] invalid.", tokens[2]);
-      }
-
-      bank.depositMoney(customer, accountName, Money.of(amount, DEFAULT_CURRENCY));
+      bank.depositMoney(
+              customerID, accountName, Money.of(amount, DEFAULT_CURRENCY));
+    } catch (AccountInvalidException ex) {
+      return String.format("FAIL: Account [%s] does not exist.", accountName);
+    }
 
       return "SUCCESS: Account credited successfully.";
+  }
+
+  /**
+   * Raises a failure message if the requested amount is inappropriate and sets the `amount` field
+   */
+  private Optional<String> invalidAmount() {
+    try {
+      amount = Double.parseDouble(tokens[2]);
     } catch (NumberFormatException ex) {
-      return String.format("FAIL: Deposit amount [%s] invalid.", tokens[2]);
-    } catch (AccountInvalidException ex) {
-      return String.format("FAIL: Account [%s] does not exist.", accountName);
+      return Optional.of(String.format("FAIL: Deposit amount [%s] invalid.", tokens[2]));
     }
+    if (amount <= 0)
+      return Optional.of(String.format("FAIL: Deposit amount [%s] invalid.", tokens[2]));
```

Comment:
I like to assign the tokens to meaningful names, because otherwise it's really easy to make a mistake, given how many times it's used.

Reply:
Yeh, good idea. Will do :+1: